### PR TITLE
Introduce pooled view buffer

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by sebastienros** [comment]
Fixes #8991

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                _pooledBuilder.Dispose();
+                _pooledBuilder = null;
+                _builder = null;
+
+                if (_previousPooledBuilders != null)
+                {
+                    foreach (var pooledBuilder in _previousPooledBuilders)
+                    {
+                        pooledBuilder.Dispose();
+                    }
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs:48]: Maybe 

    _previousPooledBuilders = null;

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            _previousPooledBuilders.Add(_pooledBuilder);
+
+            _pooledBuilder = StringBuilderPool.GetInstance();
+            _builder = _pooledBuilder.Builder;
+            return _builder;
+        }
+
+        // Invoked when used as TextWriter to intercept what is supposed to be written
+        public override void Write(string value)
+        {
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs:65]: Maybe

            if (value == null || value.Length == 0)
            {
                return;
            }

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            if (_builder.Length >= _builder.Capacity)
+            {
+                AllocateBuilder();
+            }
+
+            _builder.Append(value);
+        }
+
+        public override void Write(char[] buffer)
+        {
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs:100]: Maybe 

            if (buffer == null || buffer.Length == 0)
            {
                return;
            }

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                        AllocateBuilder();
+                    }
+
+                    index += sizeToCopy;
+                } while (index < count);
+            }
+        }
+
+        public override void Write(ReadOnlySpan<char> buffer)
+        {
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/ViewBufferTextWriterContent.cs:155]: Maybe 

            if (buffer == null || buffer.Length == 0)
            {
                return;
            }

**Pull request merged**